vllm_ascend/ops/layer_shard_linear.py

```diff
@@ -0,0 +1,313 @@
+        input_size: int,
+        output_size: int,
+        bias: bool = True,
+        skip_bias_add: bool = False,
+        params_dtype: Optional[torch.dtype] = None,
+        quant_config: Optional[QuantizationConfig] = None,
+        prefix: str = "",
+        *,
+        return_bias: bool = True,
+        cluster_name: str,
+        group: GroupCoordinator,
+        start_layer: int,
+        end_layer: int,
+        layer_idx: int,
+        prefetch_step: int = 0,
+    ):
+        self.input_size = input_size
+        self.output_size = output_size
+        self.output_partition_sizes = [output_size]
+        super().__init__(input_size,
+                         output_size,
+                         skip_bias_add,
+                         params_dtype,
+                         quant_config,
+                         prefix,
+                         return_bias=return_bias)
+        assert self.quant_method is not None
+        self.quant_method.create_weights(
+            layer=self,
+            input_size_per_partition=self.input_size,
+            output_partition_sizes=[self.output_size],
+            input_size=self.input_size,
+            output_size=self.output_size,
+            params_dtype=self.params_dtype,
+            weight_loader=self.weight_loader)
+        if bias:
+            self.bias = Parameter(
+                torch.empty(self.output_size, dtype=params_dtype))
+            set_weight_attrs(self.bias, {
+                "output_dim": 0,
+                "weight_loader": self.weight_loader,
+            })
+        else:
+            self.register_parameter("bias", None)
+
+        self.layer_idx = layer_idx
+        self.cluster = register_layer_to_cluster(
+            name=cluster_name,
+            group=group,
+            start_layer=start_layer,
+            end_layer=end_layer,
+            prefetch_step=prefetch_step,
+            layer_idx=layer_idx,
+            layer=self,
+        )
+
+    def weight_loader(self, param: Parameter, loaded_weight: torch.Tensor):
+        # Skip loading matrix weight when not intended to be stored on this device.
+        if param is self.weight and not self.cluster.is_source(self.layer_idx):
+            return
+        assert not getattr(param, "is_gguf_weight", False)
+        assert not getattr(param, "is_gguf_weight_type", False)
+        # If the weight on disk does not have a shape, give it one
+        # (such scales for AutoFp8).
+        if len(loaded_weight.shape) == 0:
+            loaded_weight = loaded_weight.reshape(1)
+        assert param.size() == loaded_weight.size(), (
+            f"Tried to load weights of size {loaded_weight.size()}"
+            f"to a parameter of size {param.size()}")
+        param.data.copy_(loaded_weight)
+
+    def post_process_after_loading_for_cluster(self):
+        self.cluster.post_process_after_loading()
+
+    def reach_layer(self):
+        self.cluster.reach_layer(self.layer_idx)
+
+    def forward(
```

Comment:
TODO: 
It should be considered to implement a default mechanism, or implicit invocation, for these two functions: `post_process_after_loading_for_cluster()`  and  `reach_layer()